tools/harness-simulation/posix/harness_dev_discovery.py

```diff
@@ -88,15 +98,27 @@ def advertise_ftd(s: socket.socket, dst, ven: str, ver: str, add: str, por: int,
             'add': f'{i}@{add}',
             'por': por,
         }
-        logging.info('Advertise: %r', info)
-        s.sendto(json.dumps(info).encode('utf-8'), dst)
+        _advertise(s, dst, info)
 
 
 def advertise_sniffer(s: socket.socket, dst, add: str, number: int):
     for i in range(number):
-        info = 'Sniffer_%d@%s' % (i, add)
-        logging.info('Advertise: %r', info)
-        s.sendto(info.encode('utf-8'), dst)
+        info = {
+            'add': add,
+            'por': i + SNIFFER_SERVER_PORT_BASE,
+        }
+        _advertise(s, dst, info)
+
+
+def initiate_sniffer(addr: str, port: int) -> subprocess.Popen:
```

Comment:
- Use a more common verb, like `start`, `create`.
- The file name is `tools/harness-simulation/posix/harness_dev_discovery.py`. Is there a better file to place the logic of starting the sniffer?

Reply:
I will use the name `start_sniffer` instead.

The filename may change to something like `discovery_and_startup.py`, the details of which can be discussed later. What the script does is to start different types of devices, and respond to the discovery messages sent by Harness.

In the following PRs, there will be an additional functionality to start OTBR Docker containers and advertise the corresponding packets. Perhaps the two steps, **discovery** and **startup**, should be split into two files. The user needs to run the discovery script and the discovery script imports the startup module, or vice versa.